financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authorize/impl/ExternalAPIConsentPersistStep.java

```diff
@@ -0,0 +1,204 @@
+import org.wso2.financial.services.accelerator.consent.mgt.extensions.common.ConsentExtensionConstants;
+import org.wso2.financial.services.accelerator.consent.mgt.extensions.common.ResponseStatus;
+import org.wso2.financial.services.accelerator.consent.mgt.extensions.internal.ConsentExtensionsDataHolder;
+import org.wso2.financial.services.accelerator.consent.mgt.service.ConsentCoreService;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Consent persist step default implementation.
+ */
+public class ExternalAPIConsentPersistStep implements ConsentPersistStep {
+
+    private ConsentCoreService consentCoreService;
+    private static final Log log = LogFactory.getLog(ExternalAPIConsentPersistStep.class);
+    private static final String CONSENT_PERSIST_DATA_OBJECT_KEY = "consent_persist_data";
+
+    public ExternalAPIConsentPersistStep() {
+        consentCoreService = ConsentExtensionsDataHolder.getInstance().getConsentCoreService();
+    }
+
+    @Override
+    public void execute(ConsentPersistData consentPersistData) throws ConsentException {
+
+        try {
+            ConsentData initialConsentData = consentPersistData.getConsentData();
+            ConsentResource consentResource;
+
+            if (initialConsentData == null) {
+                log.error("Consent data is not available");
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, AuthErrorCode.SERVER_ERROR.name(),
+                        "Consent data is not available");
+            }
+
+            if (initialConsentData.getConsentId() == null) {
+                log.error("Consent ID not available in consent data");
+                throw new ConsentException(initialConsentData.getRedirectURI(), AuthErrorCode.SERVER_ERROR,
+                        "Consent ID not available in consent data", initialConsentData.getState());
+            }
+
+            if (initialConsentData.getAuthResource() == null) {
+                log.error("Auth resource not available in consent data");
+                throw new ConsentException(initialConsentData.getRedirectURI(), AuthErrorCode.SERVER_ERROR,
+                        "Auth resource not available in consent data", initialConsentData.getState());
+            }
+
+            if (initialConsentData.getConsentResource() == null) {
+                consentResource = consentCoreService.getConsent(initialConsentData.getConsentId(), false);
+                initialConsentData.setConsentResource(consentResource);
+                consentPersistData.setConsentData(initialConsentData);
+            }
+
+            // Call external service and get updated consent persist data.
+            ConsentPersistData updatedConsentPersistData = getExternalConsentPersistData(consentPersistData);
+            ConsentData consentData = updatedConsentPersistData.getConsentData();
```

Comment:
what if someone changed the consent idclient id,from external service call,i think we should keep unmodifieble parameters list and validate those after service call,